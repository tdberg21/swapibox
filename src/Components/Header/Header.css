.header {
  display: flex;
  justify-content: space-between;
  padding-right: 50px;
  padding-left: 5%;
}

img {
  margin: 20px 0 20px 50px;
}

.show-faves-button {
  height: 30px;
  margin-top: 10px;
  background-color: #333;
  border: none;
  color: #fff;
}

h1 {
  margin-left: 50px;
}