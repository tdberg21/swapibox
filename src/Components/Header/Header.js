import React from 'react';
import './Header.css';
import PropTypes from 'prop-types';

const Header = ({ showFaves, favorites }) => {
  return (
    <div className= "header">
      <img src= "http://1000logos.net/wp-content/uploads/2017/06/Star-Wars-symbol.jpg" alt="Star Wars Logo" height="100" width="250"/>
      <button
        onClick={showFaves}
        className="show-faves-button"
      >
      View Favorites {favorites.length}
      </button>
    </div>
  );
};

Header.propTypes = {
  showFaves: PropTypes.func,
  favorites: PropTypes.array
};

export default Header;