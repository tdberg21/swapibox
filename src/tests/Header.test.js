import React from 'react';
import Header from '../Components/Header/Header.js';
import { shallow } from 'enzyme';


describe('HEADER TESTS', () => {
  let wrapper;
  let mockFaves = [];
  beforeEach(() => {
    wrapper = shallow(<Header 
                        favorites={mockFaves}
                        showFaves={jest.fn()}
                      />);
  });

  it('matches the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});