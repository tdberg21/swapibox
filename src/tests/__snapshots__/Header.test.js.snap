// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HEADER TESTS matches the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    favorites={Array []}
    showFaves={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          alt="Star Wars Logo"
          height="100"
          src="http://1000logos.net/wp-content/uploads/2017/06/Star-Wars-symbol.jpg"
          width="250"
/>,
        <button
          className="show-faves-button"
          onClick={[Function]}
>
          View Favorites 
          0
</button>,
      ],
      "className": "header",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "Star Wars Logo",
          "height": "100",
          "src": "http://1000logos.net/wp-content/uploads/2017/06/Star-Wars-symbol.jpg",
          "width": "250",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "View Favorites ",
            0,
          ],
          "className": "show-faves-button",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Array [
          "View Favorites ",
          0,
        ],
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt="Star Wars Logo"
            height="100"
            src="http://1000logos.net/wp-content/uploads/2017/06/Star-Wars-symbol.jpg"
            width="250"
/>,
          <button
            className="show-faves-button"
            onClick={[Function]}
>
            View Favorites 
            0
</button>,
        ],
        "className": "header",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "Star Wars Logo",
            "height": "100",
            "src": "http://1000logos.net/wp-content/uploads/2017/06/Star-Wars-symbol.jpg",
            "width": "250",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "View Favorites ",
              0,
            ],
            "className": "show-faves-button",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Array [
            "View Favorites ",
            0,
          ],
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
